services/appnavstate/api/src/main/kotlin/io/element/android/services/appnavstate/api/ActiveRoomsHolder.kt

```diff
@@ -53,6 +53,7 @@ class ActiveRoomsHolder {
      */
     fun removeRoom(sessionId: SessionId, roomId: RoomId) {
         val roomsForSessionId = rooms[sessionId] ?: return
+        roomsForSessionId.find { it.roomId == roomId }?.destroy()
```

Comment:
Maybe store the result of `roomsForSessionId.find { it.roomId == roomId }` to then destroy it and remove it (the line below). Here I think it is searched twice (which is not a big deal!).